### DISCUSSION TITLE
Monitor for xpanse

### DISCUSSION THREAD

**Discussion by iskey** [eclipse-xpanse/xpanse#320] [Show and tell]
@Jiajia-Wen , @swaroopar , @niuzhenguo. 
As we are going to add a monitor for xpanse.I want to show some skeleton things here. And we can discuss these design topics here.

## Here is a simple diagram of my ideas.

![xpanse-MonitorSkeleton drawio](https://user-images.githubusercontent.com/1907997/233886653-067fae9e-cb58-462b-af10-f18c31bbf04f.png)

## Some classes diagram

![image](https://user-images.githubusercontent.com/1907997/234212407-ecf2a280-a5e9-4c08-bc85-e084c48366f6.png)

**Comment by swaroopar** [comment]
@iskey Thank you for this. 
one point to be considered -> should we move the monitoring part later to a separate app? because if we have the same runtime for monitoring and for deploying services, then the load would be too high. 
Or we can control same app with different profiles. For example -> starting xpanse with monitoring profile will only handle monitoring APIs requests.

**Reply by iskey** [reply]
@swaroopar , Thank you swaroopar, separating them is a good idea.

**Comment by iskey** [comment]
An updated diagram.

![image](https://user-images.githubusercontent.com/1907997/234167584-6817519d-4e01-4436-8793-361b807e2063.png)
